### PR TITLE
Release 30 review

### PR DESCRIPTION
**Pull request by renanrodrigo**
This commit addresses review comments made in Launchpad regarding release 30

### CODE REVIEW

**Comment by renanrodrigo** [comment]
Of course we using some pytest fixture that is not present on Xenial, right?
Will fix.

File: uaclient/entitlements/repo.py [hunk 1]
```diff
@@ -183,9 +183,10 @@ def purge_kernel_check(self, package_list):
         prompt the user for confirmation before proceeding.
         """
         linux_image_versions = [
-            package.name[len("linux-image-") :]
+            m.group(1)
             for package in package_list
-            if re.match(RE_KERNEL_PKG, package.name)
+            for m in [re.search(RE_KERNEL_PKG, package.name)]
+            if m
         ]
```
Comment (orndorffgrant) [inline uaclient/entitlements/repo.py:191]: I'm not a fan of multiple loops in a list comprehension, and in this case it's kind of a hack to iterate over a list of size one just to get the match variable and check if it's None.

I recommend something like:
```python
        linux_image_versions = []
        for package in package_list:
            m = re.search(RE_KERNEL_PKG, package.name)
            if m:
                linux_image_versions.append(m.group(1))
```

File: uaclient/messages/__init__.py [hunk 2]
```diff
@@ -1991,12 +1991,12 @@ def __repr__(self):
...
 E_NOT_SETTING_PROXY_NOT_WORKING = FormattedNamedMessage(
     "proxy-not-working",
-    't.gettext("{proxy}") is not working. Not setting as proxy.',
+    t.gettext('"{proxy}" is not working. Not setting as proxy.'),
 )
 
 E_NOT_SETTING_PROXY_INVALID_URL = FormattedNamedMessage(
     "proxy-invalid-url",
-    't.gettext("{proxy}") is not a valid url. Not setting as proxy.',
+    t.gettext('"{proxy}" is not a valid url. Not setting as proxy.'),
```
Comment (orndorffgrant) [inline uaclient/messages/__init__.py:1999]: Whoops :upside_down_face: thanks

File: uaclient/files/files.py [hunk 3]
```diff
@@ -41,11 +41,15 @@ def write(self, content: str):
             if self.is_private
             else defaults.WORLD_READABLE_MODE
         )
-        if not os.path.exists(self._directory):
+        # tyy/except-ing here avoids race conditions the best
```
Comment (orndorffgrant) [inline uaclient/files/files.py:44]:
```suggestion
        # try/except-ing here avoids race conditions the best
```

**Review by orndorffgrant — APPROVED** [review]
Some minor nits, otherwise looks good - thank you!

**Comment by orndorffgrant** [comment]
But oh no the tmp_path pytest thing isn't working on xenial :(

**Comment by orndorffgrant** [comment]
It looks like other unit tests use the `tmpdir` fixture instead

File: uaclient/entitlements/repo.py (hunk 1 above)
Comment (renanrodrigo) [reply uaclient/entitlements/repo.py:191]: agreed readability++

File: uaclient/messages/__init__.py (hunk 2 above)
Comment (renanrodrigo) [reply uaclient/messages/__init__.py:1999]: anytime (:

File: uaclient/files/files.py (hunk 3 above)
Comment (renanrodrigo) [reply uaclient/files/files.py:44]: I can´t  type sorry

**Review by lucasmoura — APPROVED** [review]

**Review by bryceharrington — APPROVED** [review]
This all looks great.  The updated docs and comments read well.  Thanks for adding the thorough ESM cache and kernel test cases with such fast turnaround.  I'm not going to insist on adding the apt cache tests although would be nice in the future to strengthen the tests at that layer since it's an important interface to the underlying system, but you're right that this is already well exercised already elsewhere.

**Pull request merged**